joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/ModuleTests.scala

```diff
@@ -115,32 +115,95 @@ class ModuleTests extends RubyCode2CpgFixture {
       plays.name shouldBe "plays"
       cpg.fieldAccess.fieldIdentifier.canonicalName.l shouldBe List("plays")
     }
+  }
+
+  "Module internal structure checks with Constant defined in moudle" should {
+    val cpg = code("""
+        |module MyNamespace
+        |  MY_CONSTANT = 0
+        |end
+        |""".stripMargin)
+    "member variables structure in place" in {
+      val List(moduleInit) = cpg.method(XDefines.StaticInitMethodName).l
+      moduleInit.fullName shouldBe s"Test0.rb::program.MyNamespace.${XDefines.StaticInitMethodName}"
+      val List(myconstant) = moduleInit.call.nameExact(Operators.fieldAccess).fieldAccess.l
+      myconstant.fieldIdentifier.canonicalName.headOption shouldBe Option("MY_CONSTANT")
 
-    "Module Constant strucutre in place" ignore {
-      cpg.identifier("MY_CONSTANT").l.size shouldBe 1
-      val List(x)      = cpg.typeDecl("MyClass").l
-      val List(member) = x.member.l
-      member.name shouldBe "MY_CONSTANT"
+      val List(myclassTd)  = cpg.typeDecl("MyNamespace").l
+      val List(myConstant) = myclassTd.member.l
+      myConstant.name shouldBe "MY_CONSTANT"
+      cpg.fieldAccess.fieldIdentifier.canonicalName.l shouldBe List("MY_CONSTANT")
     }
   }
 
-  "Module internal structure checks with Constant defined in moudle" ignore {
+  "Hierarchical module checks with constants" ignore {
     val cpg = code("""
         |module MyNamespace
-        |  MY_CONSTANT = 100
+        |  MY_CONSTANT = 0
+        |  @@plays = 0
+        |  module ChildModule
+        |    @@name = 0
+        |    MY_CONSTANT = 0
+        |  end
         |end
         |""".stripMargin)
+
     "member variables structure in place" in {
-      val List(classInit) = cpg.method(XDefines.StaticInitMethodName).l
-      classInit.fullName shouldBe s"Test0.rb::program.MyNamespace.${XDefines.StaticInitMethodName}"
-      val List(myconstant) = classInit.call.nameExact(Operators.fieldAccess).fieldAccess.l
+      val List(modInit1, modInit2) = cpg.method(XDefines.StaticInitMethodName).l
+      modInit1.fullName shouldBe s"Test0.rb::program.MyNamespace.${XDefines.StaticInitMethodName}"
+      val List(myconstantfa, playsfa) = modInit1.call.nameExact(Operators.fieldAccess).fieldAccess.l
+      myconstantfa.fieldIdentifier.canonicalName.headOption shouldBe Option("MY_CONSTANT")
+      playsfa.fieldIdentifier.canonicalName.headOption shouldBe Option("plays")
```

Comment:
Not this PR's responsibility, but I confess I'm not sure if we should loose the `@@` prefix out of class fields. Right now, how would you gather them? Only via `code`?

Reply:
@DavidBakerEffendi had made those initial changes. I probably need to discuss that with him and take a call over it.

Reply:
Going through my emails now, and yeah the parser doesn't seem to differentiate field accesses from normal statements, so that was how I initially implemented this. Perhaps I am mistaken, though? I would hope the parser context would indicate a field access.